packages/gatsby/src/schema/index.js

```diff
@@ -39,9 +39,14 @@ module.exports.build = async ({ parentSpan }) => {
   const typeConflictReporter = new TypeConflictReporter()
 
   // Ensure that user-defined types are processed last
-  const sortedTypes = types.sort(
-    type => type.plugin && type.plugin.name === `default-site-plugin`
-  )
+  const sortedTypes = [
```

Comment:
This should fix sorting on all node versions. From node 12 (11.x) stable sort is being used and booleans aren't the best way to sort.

```
types.sort( 
   type => type.plugin && type.plugin.name === `default-site-plugin` ? 1 : -1
 ) 
```

Reply:
unfortuntately this does not seem to work as expected

Reply:
it seems to work for me 🤷‍♂ Oh looks like there was an issue when sorting wasn't stable.

```
types.sort( 
   (a, type) => type.plugin && type.plugin.name === `default-site-plugin` ? -1 : 1
 ) 
```
on runkit you can switch versions.
https://runkit.com/wardpeet/5d68ca5da5888a00132fe3c7

Reply:
Thanks for the link @wardpeet!

So, the following would work, but is actually quite a bit slower than filtering:
```js
const sortedTypes = types.sort((a, b) => {
  if (a.plugin && a.plugin.name === `default-site-plugin`) return 1
  if (b.plugin && b.plugin.name === `default-site-plugin`) return -1
  return 0
})
```